io/src/csv/read_measurements.cpp

```diff
@@ -48,10 +48,31 @@ void read_measurements(measurement_reader_output& out,
         }
 
         // Construct the measurement object.
-        const traccc::measurement meas{
-            point2{iomeas.local0, iomeas.local1},
-            variance2{iomeas.var_local0, iomeas.var_local1},
-            detray::geometry::barcode{iomeas.geometry_id}, link};
+        traccc::measurement meas;
+        std::array<typename transform3::size_type, 2u> indices{0u, 0u};
+        meas.meas_dim = 0u;
+
+        for (unsigned int ipar = 0; ipar < 2u; ++ipar) {
+            if (((iomeas.local_key) & (1 << (ipar + 1))) != 0) {
+
+                switch (ipar) {
+                    case e_bound_loc0: {
+                        meas.local[0] = iomeas.local0;
+                        meas.variance[0] = iomeas.var_local0;
+                        indices[meas.meas_dim++] = ipar;
+                    }; break;
+                    case e_bound_loc1: {
+                        meas.local[1] = iomeas.local1;
+                        meas.variance[1] = iomeas.var_local1;
+                        indices[meas.meas_dim++] = ipar;
+                    }; break;
+                }
+            }
+        }
```

Comment:
What does this do? It checks if `iomeas.local_key` has the second bit set when `ipar` is 0 (I guess when `iomeas.local_key` is 2 or 6. Are other cases possible, e.g. can it be 3?) and has the third bit set when `ipar` is 1, so either 4 or 6. For either case it then checks if `ipar` is 0 or 1 (are we expecting `e_bound_0` and `e_bound_1` to switch values?) and then I start to get lost because it also starts counting how often the correct bit was encountered in `meas.meas_dim`, i.e. once if `iomeas.local_key` is 2 or 4 and twice if it was 6?
I get that this has to cover many possible combinations: Can this be done in way that is maybe longer but more readable or could you draw a table in the comments on what the expected behaviour is?

Reply:
First, this is from ACTS Core and not my design. 
My interpretation might have an error but here it is:

Local key is a 8 bit char and first and last bit are dummy value. 2 - 7th bits are for 6 bound track parameters.

Ex1: 0000010 or 2 -> meas dim = 1 and bound_loc0 active -> strip or wire detector 
Ex2: 0000110 or 6 -> meas dim = 2 and bound_loc1 and bound_loc2 active -> pixel
Ex3: 0000100 or 4 -> meas dim = 1 and bound_loc2 active ->  annulus where we measure phi

Reply:
Ok, I see! Might be good to have that in the comments somewhere?

Reply:
> I get that this has to cover many possible combinations

I thought that too, but I don't see any real use case with more combinations. What current implementation provides will be enough. So I would postpone engineering this further.

Reply:
> Ok, I see! Might be good to have that in the comments somewhere?

Ok